soccer/gameplay/plays/defense/advanced_pileup.py

```diff
@@ -0,0 +1,103 @@
+import robocup
+import constants
+import play
+import skills
+import tactics
+import main
+import evaluation.position
+import math
+import random
+
+
+class AdvancedPileup(play.Play):
+    def __init__(self):
+        super().__init__(continuous=False)
+        ball = main.ball().pos
+        radius = constants.Ball.Radius
+        range = 20 * radius
+
+        c, pos = evaluation.position.ball_vicinity(range)
+
+        pos1 = None
+        pos2 = None
+        their_end = (length * 7) / 8
+        our_end = length / 8
+        left = base_width + (width / 3)
+        right = base_width + (width * 2) / 3
```

Comment:
Does this work? Where is base_width coming from?

Reply:
These were the same stipulations I used in the basic pileup, and they worked fine. I removed base_width (which was just a variable representing the left-most x value in basic pileup) and just used the left and right ends of the field as benchmarks.